baremaps-postgres/src/test/java/com/baremaps/postgres/PostgisPluginTest.java

```diff
@@ -75,4 +73,90 @@ void test() {
     assertEquals(record.getLineString(), result.get(0).getLineString());
     assertEquals(record.getPolygon(), result.get(0).getPolygon());
   }
+
+  @Test
+  @Tag("integration")
+  void beanMapper() {
+    var id = "1";
+    var type = "Feature";
+    var geometry = new GeometryFactory().createPoint(new Coordinate(1, 1));
+    var properties = Map.of("k1", "v1", "k2", "v2");
+    var feature = new Feature(id, type, geometry, properties);
+
+    Jdbi jdbi = Jdbi.create(TestConstants.DATABASE_URL).installPlugin(new PostgisPlugin());
+    List<Feature> features =
+        jdbi.withHandle(
+            handle -> {
+              var createExtension = "CREATE EXTENSION IF NOT EXISTS hstore;";
+              handle.execute(createExtension);
+
+              var createTable = "CREATE TABLE feature (id text, type text, geometry geometry, properties hstore)";
+              handle.execute(createTable);
+
+              var insert = "INSERT INTO feature (id, type, geometry, properties) VALUES (:id, :type, :geometry, :properties)";
+              handle.createUpdate(insert)
+                  .bindBean(feature)
+                  .execute();
+
+              var select = "SELECT * FROM feature ORDER BY id";
+              return handle
+                  .createQuery(select)
+                  .mapToBean(Feature.class)
+                  .list();
+            });
+
+    assertEquals(feature.getId(), features.get(0).getId());
+    assertEquals(feature.getType(), features.get(0).getType());
+    assertEquals(feature.getGeometry(), features.get(0).getGeometry());
+    assertEquals(feature.getProperties(), features.get(0).getProperties());
+  }
+
+  @Test
+  @Tag("integration")
+  void rowMapper() {
+    var id = "id";
+    var gid = "gid";
+    var type = "type";
+    var title = "title";
+    var description = "description";
+    var area = 1.5d;
+    var point = new GeometryFactory().createPoint(new Coordinate(1, 1));
+    var geom = new GeometryFactory().createPoint(new Coordinate(2, 2));
+
+    Jdbi jdbi = Jdbi.create("jdbc:tc:postgis:13-3.1:///test").installPlugin(new PostgisPlugin());
+    List<Feature> features =
+        jdbi.withHandle(
+            handle -> {
+              var createExtension = "CREATE EXTENSION IF NOT EXISTS hstore;";
```

Comment:
As testcontainers creates a fresh database for every test execution, I have to execute this query. In production, I assume that the hstore extension has already been initialized.

Reply:
I think it will nice to avoid HStore usage in Feature object and perhaps separate Feature, FeatureMapper in another maven module. So you will be able to use it with other DB as H2GIS ;-) or GeoPackage...
Or what about creating interfaces and do the implementation for each data provider ?

Reply:
I see, the difficulty here is to map the properties of the OGC Feature to the database column so that it works for selecting AND inserting data. When selecting data, we can get rid of the types, so it is easy to create a Map<String,String> based on the columns. When inserting data, we have the column names but we lack the types. The hstore drastically simplify the issue as it corresponds to a map. Any idea on how we could handle this elegantly?